grails-app/domain/mama/ng/scheduler/Message.groovy

```diff
@@ -0,0 +1,16 @@
+package mama.ng.scheduler
+
+class Message {
+
+    static hasOne = [schedule: Schedule]
+
+    String id = UUID.randomUUID().toString()
+    Date nextSend
+    Date dateCreated
+
+    static constraints = {
+        id generator:'assigned'
+        schedule(nullable: false)
+        nextSend(min: new Date(), nullable: false)
```

Comment:
dateCreated is a special character that grails fills in for you. new Date() means that the min value for next send is the date when the object is created. ie nextSend cannot be in the past.